module-03-react/src/components/Modal.jsx

```diff
@@ -0,0 +1,60 @@
+import { useEffect, useState } from 'react';
+import axios from "axios";
+
+import '/src/styles/components.css'
+
+export default function Modal({ img, openModal, choosedCard, setOpenModal }) {
+  if (!openModal || !choosedCard.url.includes("people")) return null;
+  const [films, setFilms] = useState([]);
+
+  useEffect(() => {
+    if (choosedCard && choosedCard.films) {
+      const getAPIf = async () => {
+        const filmPromises = choosedCard.films.map((filmUrl) => axios.get(filmUrl));
+        const responses = await Promise.all(filmPromises);
+        const filmData = responses.map((response) => response.data);
+        setFilms(filmData);
+      }
+      getAPIf();
+    }
+  }, [choosedCard]);
+  const [imgLink, setImgLink] = useState();
+  useEffect(() => {
+    if (img instanceof Promise) {
```

Comment:
The `img` prop handling indicates a design issue where component data flow isn't properly managed. Passing a Promise directly as a prop creates inconsistent component behavior and makes testing difficult. The Modal component shouldn't need to check if `img` is a Promise or handle Promise resolution.

Instead, resolve the Promise in the parent component (Gallery) before setting state

This approach maintains proper unidirectional data flow and keeps Promise handling logic where it belongs